### PR TITLE
Story 8

### PR DESCRIPTION
**Pull request by mfaisalnazari**
user story 6,7 and 8 complete and merge conflicts resolved.

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      
+      @item = Item.find(params[:id])
+
+    end
+
+    def update
+      merchant = Merchant.find(params[:merchant_id])
+      item = Item.find(item_params[:id])
+        item.update(item_params)
+        flash[:notice] = 'Item successfully updated.'
```
Comment (odellmac4) [inline app/controllers/items_controller.rb:23]: nice use of the flash message!

File: app/views/items/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+
+<% if flash[:notice] %>
+  <div class="flash notice"><%= flash[:notice] %></div>
```
Comment (odellmac4) [inline app/views/items/show.html.erb:3]: what is this if statement doing?

File: spec/features/items/show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      expect(page).to have_content(item1.unit_price)
+
+      expect(current_path).to eq("/merchants/#{merchant_1.id}/items/#{item1.id}")
+      
+
+    end
+
+    
+    it "can update specific merchant items" do
+        visit merchant_items_path(merchant_id: merchant_1.id)
```
Comment (odellmac4) [inline spec/features/items/show_spec.rb:25]: can we change this to read merchant_items_path(merchant) before merging?

Comment (hoadam) [reply spec/features/items/show_spec.rb:25]: I think both work, Odell's path is more clean. It is just what you prefer

File: spec/features/admin/invoices/index_spec.rb [hunk 4]
```diff
@@ -1,32 +1,32 @@
...
-            expect(page).to have_content(@invoice_2.id.to_s)
-            # Each id links to the admin invoice show page
-            expect(page).to have_link(@invoice_1.id.to_s)
-            expect(page).to have_link(@invoice_2.id.to_s)
+#         it 'shows a clickable list of all invoice ids in the system' do
+#             #As an admin,
+#             #When I visit the admin Invoices index (/admin/invoices)
+#             visit admin_invoices_path
+#             # Then I see a list of all Invoice ids in the system
+#             expect(page).to have_content(@invoice_1.id.to_s)
```
[the remark this answers is not in the document]
Comment (hoadam) [reply spec/features/admin/invoices/index_spec.rb:17]: this is Nico's

File: app/views/items/show.html.erb (hunk 2 above)
Comment (hoadam) [reply app/views/items/show.html.erb:3]: same question like Odell

Comment (hoadam) [reply app/views/items/show.html.erb:3]: I think if is optional. but I think it would be easy to read with if.

**Pull request merged**